src/image.cpp

```diff
@@ -84,9 +84,65 @@ void image::resize(const cv::Mat& input, cv::Mat& output, const cv::Size2i& size
     }
 }
 
-void image::convert_mix_channels(vector<cv::Mat>& source,
-                                 vector<cv::Mat>& target,
-                                 vector<int>&     from_to)
+void image::resize_short(const cv::Mat& input, cv::Mat& output, const int target_size)
+{
+    auto percent = static_cast<float>(target_size) / std::min(input.cols, input.rows);
+
+    auto resized_width  = static_cast<int>(std::round(input.cols * percent));
+    auto resized_height = static_cast<int>(std::round(input.rows * percent));
+    cv::resize(input, output, cv::Size2i(resized_width, resized_height), 0, 0, CV_INTER_LINEAR);
```

Comment:
this code is almost the same as src/augment_image.cpp : line 193. May be we can create separate function to avoid copy past.